### PR TITLE
glibc: update to 2.39.

### PR DESCRIPTION
**Pull request by oreo639**
#### Testing the changes
- I tested the changes in this PR: **briefly**

[ci skip]

Depends on: https://github.com/void-linux/void-packages/pull/49146
See: https://github.com/void-linux/void-packages/issues/49162

No major packaging changes indicated by upstream:
https://sourceware.org/glibc/wiki/Release/2.39
https://lists.gnu.org/archive/html/info-gnu/2024-01/msg00017.html

### CODE REVIEW

**Comment by ahesford** [comment]
I merged #49146 and then did some tests to confirm that glibc would actually build on musl masterdirs without the `localedef` dependency. In 4ed2d581049 I added a guard that will allow glibc to be built on musl without the `C.UTF-8` locale. Please improve on that logic here if you think there is a better option.

**Comment by oreo639** [comment]
> I added a guard that will allow glibc to be built on musl without the C.UTF-8 locale.

Thanks, I added an extra condition in this PR so that if cross compiling to/from the same architecture, it will do the `make -C localedata ...` (since the newly built binary will be run with the newly built glibc).

File: srcpkgs/glibc/template [hunk 1]
```diff
@@ -133,7 +128,7 @@ do_install() {
 	( cd build && make install_root=${DESTDIR} install )
 
 	# The C.UTF-8 locale should always be available
-	if [ -z "${CROSS_BUILD}" ]; then
+	if [ -z "${CROSS_BUILD}" ] || [ "${XBPS_MACHINE%-*}" = "${XBPS_TARGET_MACHINE%-*}" ]; then
```
Comment (ahesford) [inline srcpkgs/glibc/template:131]: You could just drop the `[ -z "${CROSS_BUILD}" ]` test here.

Comment (oreo639) [reply srcpkgs/glibc/template]: Alright.

**Comment by oreo639** [comment]
Merged d52e080941839e39c5425782af4f4b02f141a586

**Pull request closed without merging**